### PR TITLE
Fix combobox closes prematurely

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://github.com/adobe/react-spectrum/issues/5653

This should be safe as inputs shouldn't have anything inside them causing scrolling except for the cursor.
I think we want this as well because users should be able to navigate the entire input while combobox is open.
Other popovers block scrolling entirely, so shouldn't be an issue.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

File: packages/react-aria-components/stories/combobox-reproductions.css [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    &[data-selected] {
+      font-weight: 600;
+      background: unset;
+      color: var(--text-color);
+
+      &::before {
+        content: "✓";
+        content: "✓" / "";
+        alt: " ";
+        position: absolute;
```
Comment (ktabors) [inline packages/react-aria-components/stories/combobox-reproductions.css:82]: We don't care that scrolling the popover causes the selected indicator to scroll too, right? This is just for testing. 
image

**Review by ktabors — APPROVED** [review]
I recall us checking that this doesn't happen in v3, right? Are there any edge cases we should test for regressions? I haven't thought of anything this would break, it should theoretically only make things better.

File: packages/react-aria-components/stories/combobox-reproductions.css (hunk 1 above)
Comment (snowystinger) [reply packages/react-aria-components/stories/combobox-reproductions.css:82]: o, yeah, no need to worry about that
honestly, we can remove the story as the test does cover it. but I figured some people would like to see it in action to confirm this is the only thing that was happening

**Review by LFDanLu — APPROVED** [review]
LGTM, verified the behavior before and after. Happy to have the story removed in favor of the test, but not too bothered about keeping it in as is either
